runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -571,6 +577,10 @@ private Object applyUpdateUnsafe(SMREntry entry, long timestamp) {
                 Address.isAddress(entry.getGlobalAddress()) ? entry.getGlobalAddress() : "OPT",
                 entry.getSMRArguments());
 
+        if (SMREntry.isNOP(entry)) {
```

Comment:
Can you move this to the caller? if its a no-op, then you skip calling applyUpdateUnsafe

Reply:
Actually, if `StreamViewSMRAdapter` filter's these holes out then you dont need any changes in the VLO.

Reply:
previous and current will return an empty list<SMREntry>, so need to change that while loop a little bit.